### PR TITLE
[5G: MVC][AMF] Amf Server Service changes

### PR DESCRIPTION
**Pull request by ronit-kumar-acl**
## Summary

AMF Server service changes in existing files so as to enable the AMF server.

## Test Plan

Unit Test logs: [amf_server_unit_test.log](https://github.com/magma/magma/files/5383827/amf_server_unit_test.log)

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/oai/tasks/grpc_service/CMakeLists.txt [hunk 1]
```diff
@@ -16,8 +16,8 @@ generate_cpp_protos("${ORC8R_CPP_PROTOS}" "${PROTO_SRCS}"
   "${PROTO_HDRS}" ${ORC8R_PROTO_DIR} ${ORC8R_OUT_DIR})
 
 # SPGW
-set(SPGWSRV_LTE_CPP_PROTOS spgw_service subscriberdb policydb mobilityd)
-set(SPGWSRV_LTE_GRPC_PROTOS spgw_service)
+set(SPGWSRV_LTE_CPP_PROTOS spgw_service subscriberdb policydb mobilityd session_manager)
+set(SPGWSRV_LTE_GRPC_PROTOS spgw_service session_manager)
```
Comment (uri200) [inline lte/gateway/c/oai/tasks/grpc_service/CMakeLists.txt:20]: out of curiosity, but why is session_manager needed in this set now and it wasn't before?

Comment (ronit-kumar-acl) [reply lte/gateway/c/oai/tasks/grpc_service/CMakeLists.txt:20]: Hi correct @uri200 
We have now segregated AMF and SPGW section and currently it looks like this, I'll push the changes.
hope this makes sense ?

#AMF
set(AMFSRV_LTE_CPP_PROTOS subscriberdb session_manager)
set(AMFSRV_LTE_GRPC_PROTOS session_manager)
generate_cpp_protos("${AMFSRV_LTE_CPP_PROTOS}" "${PROTO_SRCS}"
  "${PROTO_HDRS}" ${LTE_PROTO_DIR} ${LTE_OUT_DIR})
generate_grpc_protos("${AMFSRV_LTE_GRPC_PROTOS}" "${PROTO_SRCS}"
  "${PROTO_HDRS}" ${LTE_PROTO_DIR} ${LTE_OUT_DIR})

#SPGW
set(SPGWSRV_LTE_CPP_PROTOS spgw_service subscriberdb policydb mobilityd)
set(SPGWSRV_LTE_GRPC_PROTOS spgw_service)
generate_cpp_protos("${SPGWSRV_LTE_CPP_PROTOS}" "${PROTO_SRCS}"
  "${PROTO_HDRS}" ${LTE_PROTO_DIR} ${LTE_OUT_DIR})
generate_grpc_protos("${SPGWSRV_LTE_GRPC_PROTOS}" "${PROTO_SRCS}"
  "${PROTO_HDRS}" ${LTE_PROTO_DIR} ${LTE_OUT_DIR})``

`

**Review by themarwhal — APPROVED** [review]
SMF changes looks good

**Review by uri200 — APPROVED** [review]

**Review by ulaskozat — APPROVED** [review]
LGTM

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+#include "lte/protos/session_manager.pb.h"
+#include "lte/protos/spgw_service.pb.h"
+#include <folly/IPAddress.h>
+
+extern "C" {
+#include "spgw_service_handler.h"
+#include "log.h"
+}
+#include "AmfServiceImpl.h"
```
Comment (ardzoht) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]: header for the cpp file should be the first include

**Review by ardzoht — APPROVED** [review]
approved with one minor comment

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp (hunk 2 above)
Comment (ronit-kumar-acl) [reply lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]: Sure

**Comment by themarwhal** [comment]
Hi @ronit-kumar-acl ! Please fix the DCO check failure by running `git commit --amend --signoff` & force pushing again.

**Comment by gurrapualt** [comment]
This code pulled as part of #3366 request, so closing this.

**Pull request closed without merging**